plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -125,9 +130,34 @@ impl FirmwareManager {
                 Some(((child_id, op_id), _)) = self.operation_timer.next_timed_out_entry() => {
                     let failure_reason = format!("Child device {child_id} did not respond within the timeout interval of {}sec. Operation ID={op_id}",
                         self.timeout_sec.as_secs());
-                    info!(failure_reason);
+                    error!(failure_reason);
                     self.fail_operation_in_cloud(&child_id, Some(&op_id), failure_reason).await?;
                 }
+                Some(download_result) = self.download_res_rcvr.next() => {
+                    let req_in_progress = self.req_in_progress.take().expect("infallible");
+                    let smartrest_request = req_in_progress.0;
+                    let operation_id = req_in_progress.1;
```

Comment:
Readability can be improved using a struct rather than a tuple for `req_in_progress: Option<(SmartRestFirmwareRequest, String)>`

Reply:
Obsolete as I'm not using a tuple anymore.